### PR TITLE
feat(spanner): enable client to server compression

### PR DESCRIPTION
**Pull request by rahul2393**
Enable compression of network traffic from client to server.

### CODE REVIEW

File: spanner/client.go [hunk 1]
```diff
@@ -263,13 +276,17 @@ func NewClientWithConfig(ctx context.Context, database string, config ClientConf
...
-func allClientOpts(numChannels int, userOpts ...option.ClientOption) []option.ClientOption {
+func allClientOpts(numChannels int, compression string, userOpts ...option.ClientOption) []option.ClientOption {
 	generatedDefaultOpts := vkit.DefaultClientOptions()
 	clientDefaultOpts := []option.ClientOption{
 		option.WithGRPCConnectionPool(numChannels),
 		option.WithUserAgent(fmt.Sprintf("spanner-go/v%s", internal.Version)),
 		internaloption.EnableDirectPath(true),
 	}
+	if compression == "gzip" {
+		clientDefaultOpts = append(clientDefaultOpts, option.WithGRPCDialOption(grpc.WithDefaultCallOptions(
```
Comment (olavloite) [inline spanner/client.go:287]: Would it be possible to add a test with the mock Spanner server that verifies that we actually receive this header?

**Review by olavloite — APPROVED** [review]
LGTM (with a potential request for a test), assuming that it has been verified that the setting actually has an effect (e.g. by looking at the actual number of bytes sent/received)

File: spanner/client.go (hunk 1 above)
Comment (rahul2393) [reply spanner/client.go]: Added the unit test for checking the request headers, also adding screenshot to verify all the grpc requests will have the call option for setting compression 
Screenshot 2023-05-19 at 11 55 13 AM

Comment (olavloite) [reply spanner/client.go]: Thanks for the extra info and adding the test. LGTM!

**Pull request merged**